Add Guest Role documentation (#895)
portworx/pxdocs: content/reference/CLI/role.md

```

This token will now allow access to whichever services and APIs were defined in the custom role `test.view`.

## Updating the System Guest role

Updating the `system.guest` role will change how Portworx authorization handles users that do not pass a token to any Portworx operation or SDK call. For more information, see the [Guest Role overview.](/concepts/authorization/overview#guest-access). 

### Disabling the System Guest role

To disable the system.guest role, perform the following steps: 

1. On one of your Portworx nodes, create the following `role.json` file:
    

>   ```text

Reviewer: @ggriffiths Sorry, I think you'll need another level of indentation under these. They require two tabs.
<img width="500" alt="image" src="https://user-images.githubusercontent.com/53194370/90576924-5531bb80-e174-11ea-83cb-fafacd4dabfe.png">

```

This token will now allow access to whichever services and APIs were defined in the custom role `test.view`.

## Updating the System Guest role

Updating the `system.guest` role will change how Portworx authorization handles users that do not pass a token to any Portworx operation or SDK call. For more information, see the [Guest Role overview.](/concepts/authorization/overview#guest-access). 

### Disabling the System Guest role

> To disable the system.guest role perform the following steps: 

Reviewer: ```suggestion
To disable the system.guest role, perform the following steps: 
```

1. On one of your Portworx nodes, create the following `role.json` file:
    

> ```text

Reviewer: These sections need to be indented to nest under the numbered list steps

Author: Fixed this

{
"name": "system.guest",
"rules": [
    {
    "services": [
        "!*",
    ],
    "apis": [
        "!*"
    ]
    }
}
```

2. Run the following `pxctl role update` command, specifying your `role.json` config:

```text
pxctl role update --role-config role.json
```

```text
{
  "name": "system.guest",
  "rules": [
    {
      "services": [
        "!*",
      ],
      "apis": [
        "!*"
      ]
    }
}
```

> After that, run the following command with your role.json config:

Reviewer: I think this is a duplicate of step 2, isn't it?

Author: Yes, removed this

```text
pxctl role update --role-config role.json
```

### Re-enabling the system.guest role

> To re-enable the system.guest role to it's default behavior peform the following steps:

Reviewer: ```suggestion
To re-enable the `system.guest` role to its default behavior, perform the following steps:
```

> 1. Create the following `role.json`:

Reviewer: ```suggestion
1. Create the following `role.json` file:
```

> ```text

Reviewer: these blocks also need to be indented

{
  "name": "system.guest",
  "rules": [
    {
      "services": [
        "mountattach",
        "volume",
        "cloudbackup",
        "migrate"
      ],
      "apis": [
        "*"
      ]
    },
    {
      "services": [
        "identity"
      ],
      "apis": [
        "version"
      ]
    },
    {
      "services": [
        "cluster",
        "node"
      ],
      "apis": [
        "inspect*",
        "enumerate*"
      ]
    }
  ]
}
```

> 2. Run the following command with your role.json config:

Reviewer: ```suggestion
2. Run the following `pxctl role update` command, specifying your `role.json` config:
```

```

This token will now allow access to whichever services and APIs were defined in the custom role `test.view`.

## Updating the System Guest role

Updating the `system.guest` role will change how Portworx authorization handles users that do not pass a token to any Portworx operation or SDK call. For more information, see the [Guest Role overview.](/concepts/authorization/overview#guest-access). 

### Disabling the System Guest role

> To disable the system.guest role, create the following `role.json`:

Reviewer: These steps can be broken into a numbered list. Basically any task that involves more than a single step should be numbered.

My escape characters didn't work here for the triple backticks. 

```suggestion
To disable the system.guest role perform the following steps: 

1. On one of your Portworx nodes, create the following `role.json` file:
    
    \```text
    {
    "name": "system.guest",
    "rules": [
        {
        "services": [
            "!*",
        ],
        "apis": [
            "!*"
        ]
        }
    }
    \```

2. Run the following `pxctl role update` command, specifying your `role.json` config:

     \```text
     pxctl role update --role-config role.json
     \```
```

```text
{
  "name": "system.guest",
  "rules": [
    {
      "services": [
        "!*",
      ],
      "apis": [
        "!*"
      ]
    }
}
```

After that, run the following command with your role.json config:

```text
pxctl role update --role-config role.json
```

### Re-enabling the system.guest role

> To re-enable the system.guest role to it's default behavior, create the following `role.json`:

Reviewer: same comment about numbered lists as above.

```

This token will now allow access to whichever services and APIs were defined in the custom role `test.view`.

## Updating the System Guest role

Updating the `system.guest` role will change how Portworx authorization handles users that do not pass a token to any Portworx operation or SDK call. For more information, see the [Guest Role overview.](/concepts/authorization/overview#guest-access). 

### Disabling the System Guest role

To disable the system.guest role, perform the following steps: 

1. On one of your Portworx nodes, create the following `role.json` file:
    
  ```text
  {
  "name": "system.guest",
  "rules": [
      {
      "services": [
          "!*",
      ],
      "apis": [
          "!*"
      ]
      }
  }
  ```

2. Run the following `pxctl role update` command, specifying your `role.json` config:

  ```text
  pxctl role update --role-config role.json
  ```

>   ```text

Reviewer 2: @nathan-px This errors out with the following error message:

"Role update: Failed to process role definition data, invalid character ']' looking for beginning of value"

Author: Thanks, I just fixed all of the json snippets in this PR.
